Nikita/PercolationStats.java

```diff
@@ -0,0 +1,67 @@
+import edu.princeton.cs.algs4.StdRandom;
+import edu.princeton.cs.algs4.StdOut;
+import edu.princeton.cs.algs4.StdStats;
+
+
+public class PercolationStats 
+{
+    private static final double CONF_VALUE = 1.96;
+    private final double[] fractionsOfSites;
+    private final int totalTries;
+    private final double meanRes;
+    private final double stddevRes;
+    
+    public PercolationStats(int n, int trials)
+    {   
+        validateInput(n, trials);
+        fractionsOfSites = new double[trials];
+        totalTries = trials;
+        int randomRow;
+        int randomCol;
+        int[] validInterval = {1, n+1};
+        for (int i = 0; i < trials; i++) 
+        {
+            Percolation experiment = new Percolation(n);
+            while (!experiment.percolates()) 
+            {
+                randomRow = StdRandom.uniform(validInterval[0], validInterval[1]);
+                randomCol = StdRandom.uniform(validInterval[0], validInterval[1]);
+                if (!experiment.isOpen(randomRow, randomCol))
+                    experiment.open(randomRow, randomCol);
+            }
+        fractionsOfSites[i] = experiment.numberOfOpenSites() / ((double) n * n);
+        }
+        meanRes = StdStats.mean(fractionsOfSites);
+        stddevRes = StdStats.stddev(fractionsOfSites);
+    }
+    
+    private void validateInput(int gridSize, int trialsNum)
+    {
+        if ((gridSize <= 0) || (trialsNum <= 0))
+            throw new IllegalArgumentException("Should be greater than 0");   
+    }
+    
+    public double mean() 
+    { return meanRes; }
+    
+    public double stddev()
+    { return stddevRes; }
+    
+    public double confidenceLo()
```

Comment:
This is a really small nit, but there's a potential of re-calculating the same value over and over again (you calculate these every time you call this method). They can be calculated in the constructor